language/tools/vm-genesis/src/lib.rs

```diff
@@ -90,14 +91,20 @@ pub fn encode_genesis_transaction(
     ))
 }
 
-fn merge_txn_effects(
-    mut effects_1: TransactionEffects,
-    effects_2: TransactionEffects,
-) -> TransactionEffects {
-    effects_1.resources.extend(effects_2.resources);
-    effects_1.modules.extend(effects_2.modules);
-    effects_1.events.extend(effects_2.events);
-    effects_1
```

Comment:
Just realized this (the old implementation of merge_txn_effects) could also become a time bomb in the future: if the two sets of effects we're merging overlap, we'll issue more than one commands for some resources. The DB won't necessarily choke because of this, but still it's kinda nasty and similar to the extraneous deletion bug we had...

Reply:
In other words, one property that differentiates `ChangeSet` from `TransactionEffects` is that it is guaranteed by the type system that all entries are unique. I think the reason why I used `Vec` instead of `BTreeMap` for `TransactionEffects` was that I was seeing it as a final result that didn't need to be recomputed/modified. This may be true for the normal transaction flows, but now we can see there's certainly need for manipulating these effects in the realm of writeset generation, in a way similar to how you rebase and edit git commits. I think this is also something we should consider...